### PR TITLE
Create workflows

### PR DESCRIPTION
**Pull request by krober10nd**

### CODE REVIEW

**Comment by krober10nd** [comment]
@nschloe In your Github action workflows for the projects that use pybind11, you don't install pybind11 in the CI script, right? Would this still work with previous Python versions (other than 3.9)?

**Comment by nschloe** [comment]
> you don't install pybind11 in the CI script, right?

Not explicitly, right. It's declared as a dependency in pyproject.toml (because it's needed when setuptools runs).

**Comment by krober10nd** [comment]
Looks like I just have to delete the CircleCI stuff now.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        black --check .
+    - name: Install CGAL 5
+      run: |
+        sudo apt install -y libcgal-dev
+    - name: Install other dependencies
+      run: |
+        sudo apt-get install -y python3-pip
+        sudo apt install -y libopenmpi3 libopenmpi-dev openmpi-bin
+        sudo apt install -y libhdf5-dev
+        pip install .[io] --user
```
Comment (nschloe) [inline .github/workflows/testing.yml:38]: No need to install pip or seismicmesh -- tox will handle it all.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Lint with flake8
+      run: |
+        pip install flake8
+        flake8 .
+    - name: Lint with black
+      run: |
+        pip install black
+        black --check .
```
Comment (nschloe) [inline .github/workflows/testing.yml:29]: It might be worthwhile separating out the linting. Not necessary to repeat that for every Python version.

Comment (krober10nd) [reply .github/workflows/testing.yml]: okay good point. don't want overwhelm github actions.

File: .github/workflows/testing.yml (hunk 1 above)
Comment (krober10nd) [reply .github/workflows/testing.yml]: thanks, that's right. done.

**Pull request merged**